Formula/opentsdb.rb

```diff
@@ -0,0 +1,140 @@
+    (bin/"create_table.sh").write_env_script opt_pkgshare/"tools/create_table.sh", env
+
+    inreplace pkgshare/"etc/opentsdb/opentsdb.conf", "/usr/share", "#{HOMEBREW_PREFIX}/share"
+    etc.install Dir["#{pkgshare}/etc/opentsdb"]
+  end
+
+  def post_install
+    (var/"cache/opentsdb").mkpath
+    (pkgshare/"plugins").mkpath
+
+    system "#{Formula["hbase"].opt_libexec}/bin/start-hbase.sh"
+
+    system "#{bin}/create_table.sh"
+
+    system "#{Formula["hbase"].opt_libexec}/bin/stop-hbase.sh"
+  end
+
+  plist_options :manual => "tsdb tsd --config=#{HOMEBREW_PREFIX}/etc/opentsdb/opentsdb.conf "\
+                           "--staticroot=#{HOMEBREW_PREFIX}/opt/opentsdb/share/opentsdb/static/ "\
+                           "--cachedir=#{HOMEBREW_PREFIX}/var/cache/opentsdb --port=4242 "\
+                           "--zkquorum=localhost:2181 --zkbasedir=/hbase --auto-metric"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>KeepAlive</key>
+      <dict>
+        <key>OtherJobEnabled</key>
+        <string>#{Formula["hbase"].plist_name}</string>
+      </dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/tsdb</string>
+        <string>tsd</string>
+        <string>--config=#{etc}/opentsdb/opentsdb.conf</string>
+        <string>--staticroot=#{opt_pkgshare}/static/</string>
+        <string>--cachedir=#{var}/cache/opentsdb</string>
+        <string>--port=4242</string>
+        <string>--zkquorum=localhost:2181</string>
+        <string>--zkbasedir=/hbase</string>
+        <string>--auto-metric</string>
+      </array>
+      <key>WorkingDirectory</key>
+      <string>#{HOMEBREW_PREFIX}</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/opentsdb/opentsdb.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/opentsdb/opentsdb.err</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    require "socket"
+
+    cp_r (Formula["hbase"].libexec/"conf"), testpath
+    inreplace (testpath/"conf/hbase-site.xml") do |s|
+      s.gsub! /(hbase.rootdir.*)\n.*/, "\\1\n<value>file://#{testpath}/hbase</value>"
+      s.gsub! /(hbase.zookeeper.property.dataDir.*)\n.*/, "\\1\n<value>#{testpath}/zookeeper</value>"
+    end
+
+    ENV["HBASE_LOG_DIR"]  = (testpath/"logs")
+    ENV["HBASE_CONF_DIR"] = (testpath/"conf")
+    ENV["HBASE_PID_DIR"]  = (testpath/"pid")
+
+    system "#{Formula["hbase"].opt_bin}/start-hbase.sh"
+    sleep 2
+
+    pid = fork { exec "#{bin}/tsdb tsd --config=#{HOMEBREW_PREFIX}/etc/opentsdb/opentsdb.conf --staticroot=#{HOMEBREW_PREFIX}/opt/opentsdb/share/opentsdb/static/ --cachedir=#{HOMEBREW_PREFIX}/var/cache/opentsdb --port=4242 --zkquorum=localhost:2181 --zkbasedir=/hbase --auto-metric" }
```

Comment:
Try to shorten this by wrapping the line and passing the arguments individually to `exec`. Something like this should work (and avoids an intermediate shell for argument parsing/splitting):

```ruby
pid = fork do
  exec "#{bin}/tsdb", "tsd", "--config=#{etc}/opentsdb/opentsdb.conf",
                             "--staticroot=#{opt_pkgshare}/static/",
                             "--cachedir=#{var}/cache/opentsdb",
                             "--port=4242",
                             "--zkquorum=localhost:2181",
                             "--zkbasedir=/hbase",
                             "--auto-metric"
end
```

(Note that I'm using some convenience methods, like you already used in your `.plist` file, to avoid using `HOMEBREW_PREFIX`.)